src/Microsoft.AspNet.Mvc.Razor/HelperResult.cs

```diff
@@ -41,7 +44,7 @@ public Action<TextWriter> WriteAction
         /// <param name="encoder">The <see cref="IHtmlEncoder"/> to encode the content.</param>
         public virtual void WriteTo([NotNull] TextWriter writer, [NotNull] IHtmlEncoder encoder)
         {
-            _action(writer);
+            _asyncAction(writer).Wait();
```

Comment:
Other places in code we're using `.GetAwaiter().GetResult()` to rethrow without the aggregate exception